watertap/unit_models/translator_adm1_asm1.py

```diff
@@ -0,0 +1,338 @@
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality S_ND equation",
+        )
+        def eq_Snd_conc(blk, t):
+            return blk.properties_out[t].conc_mass_comp["S_ND"] == mw_n * (
+                (
+                    blk.properties_in[t].conc_mass_comp["S_I"]
+                    * blk.config.reaction_package.N_I
+                )
+                + (
+                    blk.properties_in[t].conc_mass_comp["S_aa"]
+                    * blk.config.reaction_package.N_aa
+                )
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality Xnd equation",
+        )
+        def eq_Xnd_conc(blk, t):
+            return blk.properties_out[t].conc_mass_comp["X_ND"] == (
+                mw_n
+                * (
+                    (
+                        blk.config.reaction_package.N_bac
+                        * (
+                            blk.properties_in[t].conc_mass_comp["X_su"]
+                            + blk.properties_in[t].conc_mass_comp["X_aa"]
+                            + blk.properties_in[t].conc_mass_comp["X_fa"]
+                            + blk.properties_in[t].conc_mass_comp["X_c4"]
+                            + blk.properties_in[t].conc_mass_comp["X_pro"]
+                            + blk.properties_in[t].conc_mass_comp["X_ac"]
+                            + blk.properties_in[t].conc_mass_comp["X_h2"]
+                        )
+                    )
+                    + (
+                        blk.properties_in[t].conc_mass_comp["X_I"]
+                        * blk.config.reaction_package.N_I
+                    )
+                    + (
+                        blk.properties_in[t].conc_mass_comp["X_c"]
+                        * blk.config.reaction_package.N_xc
+                    )
+                    + (
+                        blk.properties_in[t].conc_mass_comp["X_pr"]
+                        * blk.config.reaction_package.N_aa
+                    )
+                )
+                - (blk.properties_in[t].conc_mass_comp["X_I"] * blk.i_xe)
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality alkalinity equation",
+        )
+        def return_Salk(blk, t):
+            return (
+                blk.properties_out[t].alkalinity
+                == blk.properties_in[t].conc_mass_comp["S_IC"] / mw_c
+            )
+
+        self.zero_flow_components = Set(
+            initialize=["X_BH", "X_BA", "X_P", "S_O", "S_NO"]
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.zero_flow_components,
+            doc="Components with no flow equation",
+        )
+        def return_zero_flow_comp(blk, t, i):
+            return (
+                blk.properties_out[t].conc_mass_comp[i]
+                == 1e-6 * pyunits.kg / pyunits.m**3
```

Comment:
Is 1e-6 small enough? Just wondering of the possibility of any trace concentrations that could end up being on the same order of magnitude or smaller. It may not matter but figured I'd ask.